### PR TITLE
feat: [SRM-11194]: use only useGetAppdynamicsMetricDataByPathV2 

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10818)

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/AppDCustomMetricForm/AppDCustomMetricForm.tsx [hunk 1]
```diff
@@ -163,6 +164,34 @@ export default function AppDCustomMetricForm(props: AppDCustomMetricFormInterfac
     getTypeOfInput(formikValues?.completeMetricPath as string)
   )
 
+  // const debounceSubmit = debounce((): void => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/AppDCustomMetricForm/AppDCustomMetricForm.tsx:167]: do we need this commented code?

Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/AppDCustomMetricForm/AppDCustomMetricForm.tsx]: done

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/MetricChart/__tests__/MetricChart.test.tsx [hunk 2]
```diff
@@ -25,14 +25,7 @@ describe('Unit tests for MetricChart', () => {
...
       <TestWrapper>
-        <MetricChart
-          connectorIdentifier="123123"
-          tier="1231"
-          metricPath="werwer"
-          appName="sdddf"
-          baseFolder="adsf"
-          completeMetricPath={''}
-        />
+        <MetricChart connectorIdentifier="123123" appName="sdddf" completeMetricPath={''} />
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/MetricChart/__tests__/MetricChart.test.tsx:28]: NIT : completeMetricPath=""

**Review by harshilbits — APPROVED** [review]

**Review by vasuki-kalluri — APPROVED** [review]

**Pull request merged**